doc/examples/applications/plot_3d_structure_tensor.py

```diff
@@ -0,0 +1,164 @@
+
+
+def display(im3d, cmap='gray', step=3):
+    _, axes = plt.subplots(nrows=1, ncols=3, figsize=(16, 8))
+
+    it = 0
+    for ax, image in zip(axes.flatten(), im3d[::step]):
+        ax.imshow(image, cmap=cmap, vmin=v_min, vmax=v_max)
+        ax.set_title(f'Plane = {5 + it * step}')
+        ax.set_xticks([])
+        ax.set_yticks([])
+        it += 1
+
+
+display(sample)
+
+#####################################################################
+# To view the sample data in 3D, run the following code:
+
+# import plotly.graph_objects as go
+#
+# (n_Z, n_Y, n_X) = sample.shape
+# Z, Y, X = np.mgrid[:n_Z, :n_Y, :n_X]
+#
+# fig = go.Figure(
+#     data=go.Volume(
+#         x=X.flatten(),
+#         y=Y.flatten(),
+#         z=Z.flatten(),
+#         value=sample.flatten(),
+#         opacity=0.5,
+#         slices_z=dict(show=True, locations=[4])
+#     )
+# )
+# fig.show()
+
+#####################################################################
+# Compute structure tensor
+# ========================
+# About the brightest region (i.e., at X ~ 22 and Y ~ 17), we can see
+# variations (and, hence, strong gradients) along either X or Y over a typical
+# length of 3.
+
+px.imshow(
+    sample[0, :, :],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'},
+    title='Interactive view of bottom slice of sample data.'
+)
+
+#####################################################################
+# Therefore, we choose a 'width' of 3 for the window function.
+
+sigma = 3
+A_elems = feature.structure_tensor(sample, sigma=sigma)
+
+#####################################################################
+# We can then compute the eigenvalues of the structure tensor.
+
+eigen = feature.structure_tensor_eigenvalues(A_elems)
+eigen.shape
+
+#####################################################################
+# Where is the largest eigenvalue?
+
+coords = np.where(eigen == eigen.max())
+assert coords[0] == 0  # by definition
+coords
+
+#####################################################################
+# We are looking at a local property. Let us consider a tiny neighbourhood
+# of this maximum in the X-Y plane.
+
+eigen[0, coords[1], 20:23, 12:14]
+
+#####################################################################
+# If we examine the second-largest eigenvalues in this neighbourhood, we can
+# see that they have the same order of magnitude as the largest ones.
+
+eigen[1, coords[1], 20:23, 12:14]
+
+#####################################################################
+# In contrast, the third-largest eigenvalues are one order of magnitude
+# smaller.
+
+eigen[2, coords[1], 20:23, 12:14]
+
+#####################################################################
+# As expected, the region about voxel ``(Z, Y, X) = coords[1:]`` is markedly
+# anisotropic in 3D: There is an order of magnitude between the third-largest
+# eigenvalues on one hand, and the largest and second-largest eigenvalues on
```

Comment:
```suggestion
# eigenvalue on one hand, and the largest and second-largest eigenvalues on
```

Reply:
The third-largest eigenvalues are `eigen[2, coords[1], 20:23, 12:14]` (six values) mentioned above.